silx/io/utils.py

```diff
@@ -333,40 +344,51 @@ def savespec(specfile, x, y, xlabel="X", ylabel="Y", fmt="%.7g",
 
     x_array = numpy.asarray(x)
     y_array = numpy.asarray(y)
+    if y_array.ndim > 2:
+        raise IndexError("Y columns must have be packed as 1D")
 
-    if y_array.shape[0] != x_array.shape[0]:
+    if y_array.shape[-1] != x_array.shape[0]:
         raise IndexError("X and Y columns must have the same length")
```

Comment:
Oh, i see, same exception as this one. Both could be fixed, i think.

Reply:
I let Armando or Thomas decide if it needs to be fixed (or not). There are also some consistencies in the specfile part.

Reply:
For both, `ValueError` would indeed be better, but it's been `IndexError` for the last 5 years....
To me it is not worth changing it.